script/core/hint.lua

```diff
@@ -313,6 +315,7 @@ local function semicolonHint(uri, results, start, finish)
                 end
             end
         end
+
```

Comment:
Also you are adding unrelated empty line here, I suggest reverting this (amend and force push).
Otherwise this will leave a record in the git history and show extra unrelated changes when doing git blame in future.

Reply:
Putting in the blank line was intentional as it separates multiline blocks and improves readability. It helps to clarify where the loop ends.

I don't use `git blame` often, but why would someone use it on a blank line?

Reply:
> but why would someone use it on a blank line?

It's not git blaming on the blank line, it's that when doing git blame on **this file**, it will display as `separate sections`, instead of showing the whole section is modified by the previous same commit.

https://github.com/LuaLS/lua-language-server/blame/df7cad6da565243fef13d66095cf5af2400b0369/script/core/hint.lua#L318
![image](https://github.com/user-attachments/assets/097e51aa-e241-4a70-b472-2cdab9d5d952)

>  it separates multiline blocks and improves readability

You have your point. But then when git blaming this file, it shows that this empty line is added in your **fix(hint): attempt to index a nil value** commit, but clearly it isn't part of the "fix" and that's why I said "unrelated". (adding this line fixes nothing)

---

I personally will only add these kind of blank lines if I am going to modify the surrounding codes at the same time. This results in **less unrelated changes**, similar to your ** fewer characters** principle.
But I think you will want to say this is "pedantic concerns" again 😂  So it's up to you to revert it or not.

Reply:
I think this is acceptable.